### PR TITLE
create moment first step

### PR DESCRIPTION
**Pull request by martingagnon**
## 📖 Description

Trying to catch up with 🔥 @Boubalou 's commit! 💯 

After speaking with the backend maintainer, moment creation will work this way
- Ask the backend for a upload endpoint
- Upload the image to this endpoint
- Create the moment with the end point

This pull request is a work in progress that
- has the CreateMomentViewModel
- Has everything for the upload

## Next steps
- `Navigation from home to create` -> `Navigate from create to home` (Routing?)
- Multipart upload body
- Add Graphql query to create
- UI/UX

### CODE REVIEW

File: common/src/commonMain/kotlin/com/mirego/nwad/domain/CreateMomentUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.mirego.nwad.domain
+
+import com.mirego.trikot.streams.reactive.promise.Promise
+
+interface CreateMomentUseCase {
+    fun createNewMoment(createMomentData: CreateMomentData): Promise<Unit>
+}
+
+data class CreateMomentData(val jpegImageContent: ByteArray, val title: String, val parentMoment: String? = null)
```
Comment (Boubalou) [inline common/src/commonMain/kotlin/com/mirego/nwad/domain/CreateMomentUseCase.kt:9]: ParentMoment, je suis pas certain du nom. `parentMomentId`?

File: common/src/commonMain/kotlin/com/mirego/nwad/repositories/graphql/mutations/GenerateSignedUrlMutation.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        get() = """
+mutation(${'$'}filename: String!) {
+  generateSignedUrl(filename: ${'$'}filename) {
+    successful
+    messages {
+      code
+      field
+      message
+    }
+  	result {
```
Comment (Boubalou) [inline common/src/commonMain/kotlin/com/mirego/nwad/repositories/graphql/mutations/GenerateSignedUrlMutation.kt:30]: indentation off ici

**Review by Boubalou — APPROVED** [review]
Nice!

**Review by MacTHEgenius — APPROVED** [review]
All right !

File: common/src/commonMain/kotlin/com/mirego/nwad/domain/CreateMomentUseCase.kt (hunk 1 above)
Comment (martingagnon) [reply common/src/commonMain/kotlin/com/mirego/nwad/domain/CreateMomentUseCase.kt:9]: Lorsque l'on va créer un moment a partir d'un autre moment. C'est subject to change.

**Pull request merged**